### ISSUE TITLE
incorrect ordering when sorting by non-grouped column

### ISSUE THREAD

**Issue by JulianRaya** [ashl1/datatables-rowsgroup#10]
Without the plugin, I am able to set the ordering to ascending, descending or off. But for the columns in the rowGroup array, the ordering can never be in the off state.

It makes some sense that you would code it this way, since dataTables will handle the grouping of the columns for you. With the grouped columns always sorted, equal values will always be grouped together. 
But this is not ideal for many use cases.

For instance:  If you have a table like this, that lists 1 event per hour.

    $('<table>').appendTo('body').DataTable({
        columns: [{title:"Event"},{title: "Hour"}],
        rowsGroup:[0],
        data : [
            ["server started", 1],
            ["flag raised",    2],
            ["flag lowered",   3],
            ["flag raised",    4],
            ["flag lowered",   5],
            ["flag raised",    6],
            ["flag lowered",   7],
            ["server stopped", 8]
        ]
    });

When sorting by the "hour" column, I would expect the "server started" group (single row) would be moved to the top, followed by  the "flag raised" , then "flag lowered" and finally "server stopped"
The hour column would not be sorted perfectly due to the grouping of rows, but it would show which event happened first by bringing it to the top. But this does not happen because the event column is always sorted first.   In a table like this with many event types and hundreds of rows, it would be very difficult to find which event happened first. Even using the shift key, the rows are never sorted correctly.

Expected result when sorting by hour column (note the "event" column is not sorted)

    Event           | Hour
    -----------------------------
    server started	| 1
    flag raised     | 2
                    | 4
                    | 6
    flag lowered	| 3
                    | 5
                    | 7
    server stopped  | 8

Real result sorting by hour column

    Event           | Hour
    -----------------------------
    flag raised     | 2
                    | 4
                    | 6
    flag lowered	| 3
                    | 5
                    | 7
    server started	| 1
    server stopped  | 8

**Comment by ashl1** [comment]
What you want isn't possible in this plugin "by design". If you group rows in column, the column automatically will be ordered. You can only change the order between grouped rows and inside the group only for non-grouped columns.

You are wrong in real result. The result now will be

    Event (order 1) | Hour (ordered 2)
    -----------------------------
    flag lowered	| 3
                    | 5
                    | 7
    flag raised     | 2
                    | 4
                    | 6
    server started	| 1
    server stopped  | 8

The plugin is based in Datatables and want work with minimum modification in tables provided by Datatables. Thus, I use sorting and paging from Datatables. This helps the code to be clean. To provide the functionality you want I must write most of the Datatables code from the scratch:
1. sorting
1. paging
1. filtering
and others. It is easier to make such a code in ReactJS from the scratch.

**Comment by JulianRaya** [comment]
I wasn't wrong in the real result. I only clicked the "hour" header and did not click the "event" header. I purposely did not sort by the "event" column. I did it this way to show that this is a bug whether or not you choose to implement the suggested fix, which, I admit, would be annoyingly complex given the way datatables is written. This does not mean it can't be implemented in a clean manner, it would just require a lot of modifications.

Another, more simple solution would be to disallow ordering by any column but the grouped column. Or perhaps have an option to show the user a warning if they try to order by a non-grouped column without having the grouped columns ordered.
